### PR TITLE
Improve write() test coverage and fix behaviour

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: brion/spikeReport.h [hunk 1]
```diff
@@ -53,7 +53,7 @@ class SpikeReport;
  *
  * This class is not thread-safe except where stated otherwise.
  */
-class SpikeReport : public boost::noncopyable
+class SpikeReport
```
Comment (kaabimg) [inline brion/spikeReport.h]: I would prefer to keep the boost::noncopyable: It makes clear from the beginning that the class is not copyable and I don't have to scroll down to the private section to understand that.
The noncopyable boost header is anyway included from <types.h>

Comment (tribal-tec) [inline brion/spikeReport.h]: Imo we removed it in other places in favor of the C++11 support for =delete. For readability one could put the private =delete section at the top of the class definition.

Comment (kaabimg) [inline brion/spikeReport.h]: I get this point, but sometimes =delete is used for some but not all constructors/assignment operators in order. In this case you will have to check all of them to check if the type is noncopyable or not. For me, boost::noncopyable is much clearer and makes the code shorter.

Comment (eile) [inline brion/spikeReport.h]: Right, but I just made this class movable.

Comment (hernando) [inline brion/spikeReport.h]: Certainly there's nothing wrong to make this class movable as iostreams are.

**Comment by eile** [comment]
ready for review.

**Comment by kaabimg** [comment]
Checking each time that the spike vector is sorted is bad for performance. Fixing the doc is better. We can also add an assert (isSorted(spikes));

**Comment by eile** [comment]
We already iterate over each spike in append(). Doing the check there probably costs nothing.

**Comment by hernando** [comment]
Except that throwing after having written something is not exception safe. The check and the write should be independent so for big spikes sets than do not fit in cache there's obviously a cost. However I don't think were are going to write > 10^2 MBs in a single call even in multi-million cell simulations because it's biologically impossible that all fire at the same time during a reporting delta.

**Pull request merged**